base/compiler/ssair/irinterp.jl

```diff
@@ -108,11 +84,55 @@ function getindex(tpdum::TwoPhaseDefUseMap, idx::Int)
     return TwoPhaseVectorView(tpdum.data, nelems, range)
 end
 
-function concrete_eval_invoke(interp::AbstractInterpreter, ir::IRCode, mi_cache,
-    inst::Expr, mi::MethodInstance)
+struct IRInterpretationState
+    ir::IRCode
+    mi::MethodInstance
+    world::UInt
+    argtypes_refined::Vector{Bool}
+    tpdum::TwoPhaseDefUseMap
+    ssa_refined::BitSet
+    lazydomtree::LazyDomtree
+    function IRInterpretationState(interp::AbstractInterpreter,
+        ir::IRCode, mi::MethodInstance, world::UInt, argtypes::Vector{Any})
+        argtypes = va_process_argtypes(argtypes, mi)
+        argtypes_refined = Bool[!⊑(typeinf_lattice(interp), ir.argtypes[i], argtypes[i]) for i = 1:length(argtypes)]
+        empty!(ir.argtypes)
+        append!(ir.argtypes, argtypes)
+        tpdum = TwoPhaseDefUseMap(length(ir.stmts))
+        ssa_refined = BitSet()
+        lazydomtree = LazyDomtree(ir)
+        return new(ir, mi, world, argtypes_refined, tpdum, ssa_refined, lazydomtree)
+    end
+end
```

Comment:
I setup this `IRInterpretationState` that does a similar role as `InferenceState` in order to simplify the signatures of irinterp subroutines. @Keno does it seem fine to you?

Reply:
Sure, seems fine.